src/callbacks_step/analysis.jl

```diff
@@ -377,8 +384,13 @@ function (analysis_callback::AnalysisCallback)(io, du, u, u_ode, t, semi)
         println()
     end
 
-    # Calculate L2/Linf errors, which are also returned
-    l2_error, linf_error = calc_error_norms(u_ode, t, analyzer, semi, cache_analysis)
+    if :l2_error in analysis_errors || :linf_error in analysis_errors
+        # Calculate L2/Linf errors, which are also returned
+        l2_error, linf_error = calc_error_norms(u_ode, t, analyzer, semi,
+                                                cache_analysis)
+    else
+        return nothing
```

Comment:
There is more stuff like analyzing integrals below that will be skipped by this version. You should move the printing of the l2/linf errors to this `if` block and remove the `else` part here, I think.